api/src/main/java/com/seen/api/movie/MovieController.java

```diff
@@ -35,15 +36,83 @@ public Page<Movie> search(@RequestParam(value = "title", required = true) String
-    return null;
-
+  /**
+   * Delete a movie from database.
+   *
+   * @param movieId ID of the record to delete.
+   */
+  @ApiOperation(value = "Delete movie from database", response = ResponseEntity.class)
+  @ApiResponses(value = {
+          @ApiResponse(responseCode = "200", description = "Record deleted"),
+          @ApiResponse(responseCode = "400", description = "Invalid request", content = @Content),
+          @ApiResponse(responseCode = "401", description = "You are not authorized to view the resource", content = @Content),
+          @ApiResponse(responseCode = "403", description = "Access forbidden", content = @Content),
+          @ApiResponse(responseCode = "404", description = "Not found", content = @Content),
+          @ApiResponse(responseCode = "500", description = "Internal server error", content = @Content)
+  })
+  @DeleteMapping(value = "/api/movie/delete")
+  public void deleteOrder(@RequestParam(value = "id") Long movieId) {
+    movieService.deleteMovie(movieId);
   }
 
+  /**
+   * Update an existing movie in the database.
+   *
+   * @param movieId ID of the record to update.
+   *
+   * @param movieDto Movie payload.
+   *
+   * @return An object of type Movie
+   */
+  @ApiOperation(value = "Edit a movie", response = ResponseEntity.class)
+  @ApiResponses(value = {
+          @ApiResponse(responseCode = "200", description = "Record updated"),
+          @ApiResponse(responseCode = "400", description = "Invalid request", content = @Content),
+          @ApiResponse(responseCode = "401", description = "You are not authorized to view the resource", content = @Content),
+          @ApiResponse(responseCode = "403", description = "Access forbidden", content = @Content),
+          @ApiResponse(responseCode = "404", description = "Not found", content = @Content),
+          @ApiResponse(responseCode = "500", description = "Internal server error", content = @Content)
+  })
+  @PutMapping(value = "/api/movie/edit", produces = MediaType.APPLICATION_JSON_VALUE)
+  public Movie editMovie(@RequestParam(value= "id") Long movieId, @RequestBody MovieDto movieDto) {
+    Movie movie = movieConverter.toMovie(movieDto);
+    if (movieService.findMovieById(movieId).isPresent()) {
+      Movie existingMovie = movieService.findMovieById(movieId).get();
 
+      existingMovie.setTitle(movieDto.getTitle());
+      existingMovie.setDescription(movieDto.getDescription());
+      existingMovie.setDuration(movieDto.getDuration());
+      existingMovie.setReleaseDate(movieDto.getReleaseDate());
+      existingMovie.setGenre(movieDto.getGenre());
+      existingMovie.setRating(movieDto.getRating());
+      existingMovie.setRatingImdb(movieDto.getRatingImdb());
+      existingMovie.setRatingRottenTomatoes(movieDto.getRatingRottenTomatoes());
+      existingMovie.setParentalControl(movieDto.getParentalControl());
+
+      movieService.saveMovie(existingMovie);
+    }
+    return movie;
+  }
```

Comment:
1. Line 100: What's the point of it and why do it before checking if the movie exists.
2. Line 101 & 102 both request the movie entity (so you do it twice for no reason)
3. Line 101 should be `movieService.isPresent()` and the function should call `movieRepository.existsById`
4. Lines 104 to 112 should look similar to `MovieConverter.toMovie` but instead it should take as arguments the fetched movie from db and then set everything besides the ID.
5. everything should be extracted into `updateMovie()`